src/utils/translateKeys.ts

```diff
@@ -0,0 +1,24 @@
+export default function translateKeys(
+  data: unknown,
+  translate: (key: string) => string,
+): unknown {
+  if (Array.isArray(data)) {
+    return data.map(value => translateKeys(value, translate));
+  }
+
+  if (data instanceof Date) {
+    return data.toISOString();
+  }
+
+  if (!data || typeof data !== 'object') {
+    return data;
+  }
+
+  const translatedData: { [key: string]: unknown } = {};
+
+  Object.entries(data as {}).forEach(([key, value]) => {
```

Comment:
Also ughh

Reply:
I think the type will narrow correctly if you do 

```
if (typeof data !== 'object' || !data) {
  return data
}
```

switching the order. I've noticed TS is finicky on typeof object checks b/c it narrows to `{} | null` so you have to narrow to that first, then eliminate the null